### PR TITLE
Updated stock management UI for creating and viewing stock locations

### PR DESCRIPTION
**Pull request by patryllus**
chrome-capture-2024-3-17

### CODE REVIEW

File: src/stock-locations/stock-locations-table.resource.tsx [hunk 1]
```diff
@@ -46,28 +55,73 @@ export function useStockLocationPages(filter: StockOperationFilter) {
...
+    isLoading,
+    error,
     setPageSize,
     tableHeaders,
     tableRows,
   };
 }
+
+export const useLocationTags = () => {
+  const url = `ws/rest/v1/locationtag/`;
```
Comment (jabahum) [inline src/stock-locations/stock-locations-table.resource.tsx:88]: can we use `${restBaseUrl}`

File: src/stock-locations/stock-locations-table.resource.tsx [hunk 2]
```diff
@@ -46,28 +55,73 @@ export function useStockLocationPages(filter: StockOperationFilter) {
...
+};
+interface LocationName {
+  name: string;
+}
+export async function saveLocation({
+  locationPayload,
+}): Promise<FetchResponse<LocationName>> {
+  try {
+    const response: FetchResponse = await openmrsFetch(
+      `/ws/rest/v1/location/`,
```
Comment (jabahum) [inline src/stock-locations/stock-locations-table.resource.tsx:110]: here too

**Review by jabahum — CHANGES_REQUESTED** [review]
Thanks @patryllus we almost there just work on the changes requested herein

File: src/stock-locations/add-locations-form.component.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        name,
+        tags,
+      };
+      saveLocation({ locationPayload: locationbject })
+        .then(() => {
+          showToast({
+            title: t("formCreated", "Add Location"),
+            kind: "success",
+            critical: true,
+            description: `Location ${name} was created successfully.`,
```
Comment (makombe) [inline src/stock-locations/add-locations-form.component.tsx:42]: We can make description translatable

File: src/stock-locations/location-admin-form.component.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+              <Controller
+                name="name"
+                control={control}
+                rules={{ required: true }}
+                render={({ field, fieldState }) => (
+                  <>
+                    <TextInput
+                      id="location"
+                      labelText={t("location", "Location Name")}
+                      placeholder={t("locationPlaceholder", "")}
```
Comment (makombe) [inline src/stock-locations/location-admin-form.component.tsx:112]: Provide location placeholder text being translated here.

**Pull request closed without merging**